### PR TITLE
Clean -> Master

### PR DESCRIPTION
**Pull request by hoangphanusc**
Created domain.go & usecases.go
Need interafaces.go & infrastructure.go

### CODE REVIEW

**Comment by trietphm** [comment]
Where is your repository package?

File: src/usecases/usecases.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+type ProfileInteractor struct {
+	UserRepository    UserRepository
+	LibraryRepository LibraryRepository
+	GameRepository    domain.GameRepository
+	Logger            Logger
+}
+
+func (interactor *ProfileInteractor) Profile(userId, libraryId int) (string, []domain.Game, error) {
+	var games []domain.Game
+	user := interactor.UserRepository.FindById(userId)
```
Comment (trietphm) [inline src/usecases/usecases.go]: This method is not good. What if your SQL query has some problems? `FindById` should return an error.

File: src/usecases/usecases.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+	info := interactor.UserRepository.LoadInfo(user)
```
Comment (trietphm) [inline src/usecases/usecases.go]: Same https://github.com/frostablevenus/game-tracker/pull/1/files#r69858522

File: src/usecases/usecases.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	library := interactor.LibraryRepository.FindById(libraryId)
```
Comment (trietphm) [inline src/usecases/usecases.go]: Same as https://github.com/frostablevenus/game-tracker/pull/1/files#r69858522

File: src/usecases/usecases.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+func (interactor *ProfileInteractor) EditUserInfo(userId int, info string) error {
+	user := interactor.UserRepository.FindById(userId)
+	interactor.UserRepository.StoreInfo(user, info)
+	return nil
+}
+
+func (interactor *ProfileInteractor) Add(userId, libraryId, gameId int) error {
+	user := interactor.UserRepository.FindById(userId)
```
Comment (trietphm) [inline src/usecases/usecases.go]: Same as https://github.com/frostablevenus/game-tracker/pull/1/files#r69858522

File: src/usecases/usecases.go [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+	library := interactor.LibraryRepository.FindById(libraryId)
```
Comment (trietphm) [inline src/usecases/usecases.go]: Same as https://github.com/frostablevenus/game-tracker/pull/1/files#r69858522

File: src/usecases/usecases.go [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+		err := fmt.Errorf(message,
+			user.Id,
+			user.Player.Id,
+			library.Id,
+			library.Player.Id)
+		interactor.Logger.Log(err.Error())
+		return err
+	}
+
+	game := interactor.GameRepository.FindById(gameId)
```
Comment (trietphm) [inline src/usecases/usecases.go]: Same as https://github.com/frostablevenus/game-tracker/pull/1/files#r69858522

File: src/usecases/usecases.go [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+	library.Games = append(library.Games, game)
+	interactor.LibraryRepository.Store(library)
```
Comment (trietphm) [inline src/usecases/usecases.go]: Same as https://github.com/frostablevenus/game-tracker/pull/1/files#r69858522

File: src/usecases/usecases.go [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+	interactor.Logger.Log(fmt.Sprintf(
+		"User added game '%s' (#%i) to library #%i",
+		game.Name, game.Id, library.Id))
+	return nil
+}
+
+func (interactor *ProfileInteractor) Remove(userId, libraryId, gameId int) error {
+	user := interactor.UserRepository.FindById(userId)
```
Comment (trietphm) [inline src/usecases/usecases.go]: Same as https://github.com/frostablevenus/game-tracker/pull/1/files#r69858522

File: src/usecases/usecases.go [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+	library := interactor.LibraryRepository.FindById(libraryId)
```
Comment (trietphm) [inline src/usecases/usecases.go]: Same as https://github.com/frostablevenus/game-tracker/pull/1/files#r69858522

File: src/usecases/usecases.go [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+		err := fmt.Errorf(message,
+			user.Id,
+			user.Player.Id,
+			library.Id,
+			library.Player.Id)
+		interactor.Logger.Log(err.Error())
+		return err
+	}
+
+	game := interactor.GameRepository.FindById(gameId)
```
Comment (trietphm) [inline src/usecases/usecases.go]: Same as https://github.com/frostablevenus/game-tracker/pull/1/files#r69858522

File: src/usecases/usecases.go [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+	for i := range library.Games {
+		if game.Id == library.Games[i].Id {
+			library.Games = append(library.Games[:i], library.Games[i+1:]...)
+			interactor.LibraryRepository.Store(library)
```
Comment (trietphm) [inline src/usecases/usecases.go]: Same as https://github.com/frostablevenus/game-tracker/pull/1/files#r69858522

**Comment by trietphm** [comment]
Have you build your source code?

**Comment by hoangphanusc** [comment]
By source code, do you mean main.go?

What about repository package?

**Comment by hoangphanusc** [comment]
I'll work on FindById err check first

**Comment by trietphm** [comment]
All of them. 
Before merge into `master`, your source code should be runnable. I mean compile it without error. You can use `go install`, `go build` to make sure everything work well.
About `repository`, I want to see this those `repository` function, such as `interactor.GameRepository.FindById`

**Comment by hoangphanusc** [comment]
Repository functions are in interfaces.go, which I'll build next.

**Comment by trietphm** [comment]
- Naming: `clean` is not a good name, it should be something makes sense.
- One more question: `src/domain/domain.go` something's wrong? You project should be:

```
$GOPATH
    └─src   
        └─game-tracker
             ├─domain
             ├─usecases
             ├─interfaces
             ├─infrastructure
             └─utils/something else
```

**Pull request closed without merging**